snappy_test.go

```diff
@@ -322,6 +323,18 @@ func downloadTestdata(b *testing.B, basename string) (errRet error) {
 			os.Remove(filename)
 		}
 	}()
+	// Generate random data
+	if basename == "random" {
+		rng := rand.New(rand.NewSource(27354294))
+		for i := 0; i < 1<<20; i++ {
+			_, err := f.Write([]byte{byte(rng.Intn(255))})
```

Comment:
f is an *os.File, so f.Write is a syscall. I'm not talking about the rng.Intn call. On my system:

for i := 0; i < 1<<20; i++ {
	_, err := f.Write([]byte{byte(rng.Intn(256))})
	if err != nil {
		return err
	}
}

takes 1.6 seconds. Instead:

b := make([]byte, 1<<20)
for i := range b {
	b[i] = byte(rng.Intn(256))
}
if _, err := f.Write(b); err != nil {
	return err
}

takes 20ms.

It may be a one-time operation, but it's not hard to do it well.

In any case, as I said, further tests over and above what C++ does is fine, but they shouldn't be in the section marked "copied directly from [the C++]".